server-embedded/src/main/java/org/exoplatform/chat/services/mongodb/ChatMongoDataStorage.java

```diff
@@ -872,43 +872,51 @@ public RoomsBean getUserRooms(String user, List<String> onlineUsers, String filt
-        if(spaces != null) {
-          for (Object room : spaces) {
-            roomsIds.add((String) room);
+      DBObject roomsQuery = null;
+      if (TYPE_ROOM_FAVORITE.equals(roomType)) {
+        List<String> favoriteRoomsIds = userBean.getFavorites();
+        if (favoriteRoomsIds != null) {
+          roomsQuery = new BasicDBObject("_id", new BasicDBObject("$in", favoriteRoomsIds));
+        }
+      } else {
+        List<String> roomsIds = new ArrayList<>();
+        List<BasicDBObject> andList = new ArrayList<>();
+        List<BasicDBObject> orList = new ArrayList<>();
+        DBObject doc = cursor.next();
+        if (StringUtils.isBlank(roomType) || TYPE_ROOM_SPACE.equals(roomType)) {
+          BasicDBList spaces = (BasicDBList) doc.get("spaces");
+          if (spaces != null) {
+            for (Object room : spaces) {
+              roomsIds.add((String) room);
+            }
           }
         }
-      }
-      if(StringUtils.isBlank(roomType) ||  TYPE_ROOM_TEAM.equals(roomType)) {
-        BasicDBList teams = (BasicDBList) doc.get("teams");
-        if (teams != null) {
-          for (Object room : teams) {
-            roomsIds.add((String) room);
+        if (StringUtils.isBlank(roomType) || TYPE_ROOM_TEAM.equals(roomType)) {
+          BasicDBList teams = (BasicDBList) doc.get("teams");
+          if (teams != null) {
+            for (Object room : teams) {
+              roomsIds.add((String) room);
+            }
           }
         }
-      }
-      // Add spaces and teams rooms
-      orList.add(new BasicDBObject("_id", new BasicDBObject("$in", roomsIds)));
-      if(StringUtils.isBlank(roomType) || TYPE_ROOM_USER.equals(roomType)) {
-        // Add user to user rooms
-        orList.add(new BasicDBObject("users", user));
-      }
+        // Add spaces and teams rooms
+        orList.add(new BasicDBObject("_id", new BasicDBObject("$in", roomsIds)));
+        if (StringUtils.isBlank(roomType) || TYPE_ROOM_USER.equals(roomType)) {
+          // Add user to user rooms
+          orList.add(new BasicDBObject("users", user));
+        }
 
-      DBObject roomsQuery = new BasicDBObject("$and", andList);
-      List<BasicDBObject> enabledRoomOrList = new ArrayList<>();
-      enabledRoomOrList.add(new BasicDBObject("isEnabled", true));
-      enabledRoomOrList.add(new BasicDBObject("isEnabled", new BasicDBObject("$exists", false)));
-      andList.add(new BasicDBObject("$or", enabledRoomOrList));
-      andList.add(new BasicDBObject("$or", orList));
+        roomsQuery = new BasicDBObject("$and", andList);
+        List<BasicDBObject> enabledRoomOrList = new ArrayList<>();
+        enabledRoomOrList.add(new BasicDBObject("isEnabled", true));
+        enabledRoomOrList.add(new BasicDBObject("isEnabled", new BasicDBObject("$exists", false)));
+        andList.add(new BasicDBObject("$or", enabledRoomOrList));
+        andList.add(new BasicDBObject("$or", orList));
+      }
 
```

Comment:
I think the part about isenabled should be kept even for favorites. If you fav a room which is no more enabled, we should not display it. So the part about isEnabled shoud be outside the if

Reply:
Done.